packages/core/admin/admin/src/core/apis/Components.js

```diff
@@ -0,0 +1,43 @@
+import { cloneDeep } from 'lodash';
+import invariant from 'invariant';
+
+// TODO this API should be merged with the components one
+class Components {
+  components = {};
+
+  getComponent = name => {
+    invariant(name, 'A name must be provided');
+
+    return cloneDeep(this.components[name]) || null;
+  };
+
+  getComponents = () => {
+    const components = cloneDeep(this.components);
+
+    return Object.keys(components).reduce((acc, current) => {
+      acc[current] = components[current].Component;
+
+      return acc;
+    }, {});
+  };
+
+  registerComponent = component => {
+    const { name, Component } = component;
+
+    invariant(Component, 'A Component must be provided');
+    invariant(name, 'A name must be provided');
+    invariant(this.components[name] === undefined, 'A similar field already exists');
+
+    this.components[name] = { Component };
```

Comment:
will there be other keys in this objet? If not, why not to:

```js
this.components[name] = Component
```

this will simplify the `getComponents` method by only returning `this.components` instead of iterating through each of the keys